boefjes/plugins/kat_webpage_capture/main.py

```diff
@@ -0,0 +1,77 @@
+from typing import Tuple, Union, List
+
+from boefjes.job_models import BoefjeMeta
+from PIL import Image
+import io
+import os
+import docker
+import logging
+import tarfile
+
+PLAYWRIGHT_IMAGE = "mcr.microsoft.com/playwright:v1.30.0-focal"
+BROWSER = "chromium"
+
+
+def get_file_from_container(container: docker.models.containers.Container, path: str) -> bytes:
+    """Returns a file from a docker container."""
+    try:
+        stream, _ = container.get_archive(path)
+    except docker.errors.NotFound:
+        logging.warning(f"[Webpage Capture] {path} not found in container {container.short_id} {container.image.tags}")
+        return None
+
+    # Extract a file as archive stream and write to BytesIO.
+    file_as_tar_bytes = io.BytesIO()
```

Comment:
(haven't tested this) I think this block is redundant since you can pass the `stream` directly a file handle object to `tarfile.open`. This way you also won't have to read the stream twice

Reply:
It turned out to be a bit complex, but I think I have a working solution now